docs/develop/php/versioning.mdx

```diff
@@ -0,0 +1,212 @@
+toc_max_heading_level: 4
+keywords:
+  - best practices
+  - code sample
+  - deployment
+  - deployment safety
+  - deprecated patches
+  - how-to
+  - patching
+  - php
+  - php sdk
+  - version
+  - versioning
+  - workflow completion
+  - workflow history
+  - workflow transition
+tags:
+  - best-practices
+  - code-sample
+  - deployment
+  - deployment-safety
+  - deprecated-patches
+  - how-to
+  - patching
+  - php
+  - php-sdk
+  - version
+  - versioning
+  - workflow-completion
+  - workflow-history
+  - workflow-transition
+---
+
+A Temporal Workflow Definition must be deterministic.
+Temporal uses event sourcing to reconstruct Workflow state by replaying saved History Event data with the Workflow Definition code.
+This means that any incompatible update to the Workflow definition code could cause a non-deterministic
+issue if not handled correctly.
+
+## Introduction to Versioning {#introduction-to-versioning}
+
+To design for potentially long running Workflows at scale, versioning with Temporal works differently than you might expect.
+Discover more in this optional 30 minute introduction: [https://www.youtube.com/watch?v=kkP899WxgzY](https://www.youtube.com/watch?v=kkP899WxgzY)
+
+## How to use the PHP SDK Patching API {#php-sdk-patching-api}
+
+The PHP SDK's patching mechanism operates similarly to other SDKs in a "feature-flag" fashion. 
+The "versioning" API now uses the concept of "patching in" code.
+
+To understand this, you can break it down into three steps, which reflect three stages of migration:
+
+- Running `prePatchActivity` code while concurrently patching in `postPatchActivity`.
+- Running `postPatchActivity` code with deprecation markers for `step-1` patches.
+- Running only the `postPatchActivity` code.
+
+Let's walk through this process in sequence.
+
+Suppose you have an initial Workflow version called `PrePatchActivity`:
+
+```php
+#[WorkflowInterface]
+class MyWorkflow
+{
+    private $activity;
+
+    public function __construct()
+    {
+        $this->activity = Workflow::newActivityStub(
+            YourActivityInterface::class,
+            ActivityOptions::new()->withScheduleToStartTimeout(60)
+        );
+    }
+
+    #[WorkflowMethod]
+    public function runAsync()
+    {
+        $result = yield $this->activity->prePatchActivity();
+    }
+}
+```
+
+Now, you want to update your code to run `postPatchActivity` instead. This represents your desired end state.
+
+```php
+#[WorkflowInterface]
+class MyWorkflow
+{
+    // ...
+
+    #[WorkflowMethod]
+    public function runAsync()
+    {
+        $result = yield $this->activity->postPatchActivity();
+    }
+}
+```
+
+**Problem: You cannot deploy `postPatchActivity` directly until you're certain there are no more running Workflows created using the `prePatchActivity` code, otherwise you are likely to cause a non-deterministic error.**
```

Comment:
```suggestion
**Problem**: You can't deploy `postPatchActivity` directly until you're certain there are no more running Workflows created using the `prePatchActivity` code.
Otherwise you are likely to cause a non-deterministic error.
```